registry/auth/htpasswd/access.go

```diff
@@ -60,7 +53,32 @@ func (ac *accessController) Authorized(ctx context.Context, accessRecords ...aut
 		}
 	}
 
-	if err := ac.AuthenticateUser(username, password); err != nil {
+	// Dynamically parsing the latest account list
+	fstat, err := os.Stat(ac.path)
+	if err != nil {
+		return nil, err
+	}
+
+	lastModified := fstat.ModTime().String()
+	ac.mtx.Lock()
+	if ac.modtime != lastModified {
+		ac.modtime = lastModified
+
+		f, err := os.Open(ac.path)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+
+		h, err := newHTPasswd(f)
+		if err != nil {
+			return nil, err
+		}
+		ac.htpasswd = h
+	}
+	ac.mtx.Unlock()
+
+	if err := ac.htpasswd.authenticateUser(username, password); err != nil {
```

Comment:
You also need to hold the lock when you access `ac.htpasswd`.

Maybe set a local variable to the value of `ac.htpasswd` before releasing the lock above.